src/java/com/zimbra/graphql/utilities/XMLDocumentUtilities.java

```diff
@@ -75,6 +77,26 @@ public static Element executeDocument(DocumentHandler handler, ZimbraSoapContext
         return handler.handle(request, context);
     }
 
+    /**
+     * Executes a given request on a document handler.
+     *
+     * @param handler The handler to handle the request
+     * @param zsc The zimbra soap context for the request
+     * @param rctxt The request context
+     * @param request The request to execute
+     * @return The document response
+     * @throws ServiceException If there are issues executing the document
+     */
+    public static Element executeDocument(DocumentHandler handler, ZimbraSoapContext zsc,
+        RequestContext rctxt, Element request)
+        throws ServiceException {
+        final Map<String, Object> context = new HashMap<String, Object>();
+        final HttpServletRequest req = rctxt.getRawRequest();
+        context.put(SoapEngine.ZIMBRA_CONTEXT, zsc);
+        context.put(SoapServlet.SERVLET_REQUEST, req);
```

Comment:
Just add this to the pre-existing executeDocument - no need to add a new method.